common/src/main/java/org/apache/celeborn/common/network/sasl/CelebornSaslClient.java

```diff
@@ -58,6 +58,7 @@ public CelebornSaslClient(
       @Nullable Map<String, String> saslProps,
       @Nullable CallbackHandler authCallbackHandler) {
     Preconditions.checkNotNull(saslMechanism);
+    initializeSaslProviders();
     try {
       this.saslClient =
```

Comment:
Could `saslClient` be null? `createSaslClient` may return null.

Reply:
Yeah it can be null. Everywhere it is used, we do check if it is `null` or not.

Reply:
@otterc, I mean that the `CelebornSaslClient` does nothing when `saslClient` is null, therefore the `saslClient` could not be null.

Reply:
Seem SparkSaslClient has similar behavior, I suppose this is fine :)